girder/api/v1/resource.py

```diff
@@ -369,3 +374,55 @@ def copyResources(self, params):
         .errorResponse('No resources specified.')
         .errorResponse('Resource not found.')
         .errorResponse('ID was invalid.'))
+
+    @access.user
+    def mergeResources(self, params):
+        """
+        Merge the selected resources to a new parent folder. Only folder and
+        item resources can be moved with this function.
+        """
+        user, resources, parent, parentType, progress = \
+            self._prepareMoveCopyOrMerge(params, operation='merge')
+        name = params.get('name')
+        description = params.get('description', '')
+        total = sum([len(resources[key]) for key in resources])
+        with ProgressContext(progress, user=user, title='Moving resources',
+                             message='Calculating requirements...',
+                             total=total) as ctx:
+            for kind in resources:
+                model = self._getResourceModel(kind)
+                toBeMerged = {
+                    'item': [],
+                    'folder': []
+                }
+                for id in resources[kind]:
+                    doc = model.load(id=id, user=user, level=AccessType.WRITE)
```

Comment:
If we use `exc=True` in the load call, it will take care of throwing the `RestException` for you.